pkg/orchestration/controller.go

```diff
@@ -151,6 +157,25 @@ func (c *Controller) process(logger *zap.Logger, state *orch_v1.State) (conflict
 	return conflict, retriable, realBundle, err
 }
 
+func parseConfigMap(configMap *core_v1.ConfigMap) (*orch_meta.ServiceProperties, error) {
+	configMapConfigData, ok := configMap.BinaryData[orch_meta.ConfigMapConfigKey]
+	if !ok {
```

Comment:
Kind of interesting to sort it this way - usually you check preconditions, then perform the operation.

This optimizes for the happy path, I guess...